rules/rules.js

```diff
@@ -371,85 +371,92 @@ const SwitchableJSRule = function (ruleConfig) {
   }
 };
 
+/**
+ * Get rule trigger data from raw Java input and generate JavScript object.
+ * @private
+ * @param {*} input raw Java input from openHAB core
+ * @returns {rules.EventObject}
+ */
 const getTriggeredData = function (input) {
   const event = input.get('event');
+  const data = {};
 
-  if (event && Java.typeName(event.class) === 'org.openhab.core.items.events.ItemCommandEvent') {
-    return {
-      eventType: 'command',
-      triggerType: 'ItemCommandTrigger',
-      receivedCommand: event.getItemCommand().toString(),
-      oldState: input.get('oldState') + '',
-      newState: input.get('newState') + '',
-      itemName: event.getItemName(),
-      module: input.get('module')
-    };
-  }
+  // Properties of data are dynamically added, depending on their availability
 
-  const ev = event + '';
-  // log.debug("event",ev.split("'").join("").split("Item ").join("").split(" "));
-  let evArr = [];
-  if (ev.includes('triggered')) {
-    const atmp = ev.split(' triggered '); // astro:sun:local:astroDawn#event triggered START
-    evArr = [atmp[0], 'triggered', atmp[1]];
-  } else {
-    evArr = ev.split("'").join('').split('Item ').join('').split(' '); // Item 'benqth681_switch' received command ON
+  // Item triggers
+  if (input.containsKey('command')) {
+    data.receivedCommand = input.get('command').toString();
+    data.command = input.get('command').toString();
+  }
+  if (input.containsKey('oldState')) data.oldState = input.get('oldState').toString();
+  if (input.containsKey('newState')) data.newState = input.get('newState').toString();
+  if (input.containsKey('state')) {
+    data.receivedState = input.get('state').toString();
+    data.state = input.get('state').toString();
   }
 
-  const d = {
-    // size: input.size(),
-    oldState: input.get('oldState') + '',
-    newState: input.get('newState') + '',
-    state: input.get('state') + '', // this occurs on an ItemStateUpdateTrigger
-    receivedCommand: null,
-    receivedState: null,
-    receivedTrigger: null,
-    itemName: evArr[0],
-    module: input.get('module'),
-    command: input.get('command') + '' // for PWM trigger
-  };
-
-  try {
-    if (event !== null && event.getPayload()) {
-      d.payload = JSON.parse(event.getPayload());
-      log.debug('Extracted event payload {}', d.payload);
+  // Thing triggers
+  if (input.containsKey('newStatus')) data.newStatus = input.get('newStatus').toString();
+  if (input.containsKey('oldStatus')) data.oldStatus = input.get('oldStatus').toString();
+  if (input.containsKey('status')) data.status = input.get('status').toString();
```

Comment:
There are lots of repeated statements here, personally I would extract this logic to a function.

Reply:
Done, see [72f2c4](https://github.com/openhab/openhab-js/pull/136/commits/472f2c4a45fcb34ece831d0aa4f32716eb801a79)